lib/jquery.ui/jquery.ui.preview.js

```diff
@@ -48,8 +48,12 @@ $.widget( 'ui.preview', {
 	 * @protected
 	 */
 	_create: function() {
+		var hashBasedMessageProvider = new util.HashMessageProvider( this.options.messages );
 		if( this.options.messageProvider ) {
-			this.options.messageProvider.setDefaultMessages( this.options.messages );
+			this.options.messageProvider = this.options.messageProvider ? new util.CombiningMessageProvider(
```

Comment:
Half of this is dead code. `this.options.messageProvider` is checked to not be `false` in the line above.